src/components/card/card-footer/card-footer.test.tsx

```diff
@@ -0,0 +1,68 @@
+import React from "react";
+import { render, screen } from "@testing-library/react";
+import CardFooter, { CardFooterProps } from ".";
+import CardContext from "../__internal__/card.context";
+
+test("when variant prop is `transparent`, render with transparent background", () => {
+  render(
+    <CardFooter variant="transparent" data-role="card-footer">
+      <div id="non-interactive">View Stripe Dashboard</div>
+    </CardFooter>
+  );
+
+  const cardFooterElement = screen.getByTestId("card-footer");
+  expect(cardFooterElement).toHaveStyle({
+    "background-color": "transparent",
+  });
+});
+
+test.each<CardFooterProps["roundness"]>(["default", "large"])(
+  "renders with the expected border radius styling when roundness is %s",
+  (roundness) => {
+    render(
+      <CardContext.Provider
+        value={{
+          roundness,
+          spacing: "medium",
+        }}
+      >
+        <CardFooter data-role="card-footer">foo</CardFooter>
+      </CardContext.Provider>
+    );
+
+    const cardFooterElement = screen.getByTestId("card-footer");
+    expect(cardFooterElement).toHaveStyle({
+      borderBottomLeftRadius: `var(--borderRadius${
+        roundness === "default" ? "1" : "2"
+      }00)`,
+      borderBottomRightRadius: `var(--borderRadius${
+        roundness === "default" ? "1" : "2"
+      }00)`,
+    });
+  }
+);
+
+test("has the expected data attributes when they are passed in", () => {
```

Comment:
comment: again not sure if we need to renders here but there's defaults so not a huge deal to leave as is